sdk/core/azure-core/tests/perf_tests/_test_base.py

```diff
@@ -0,0 +1,354 @@
+                    self.async_transport = async_transport_types[self.args.transport]
+                except KeyError:
+                    raise ValueError(f"Invalid async transport:{self.args.transport}\n Valid options are:\n- aiohttp\n- requests\n")
+
+        self.error_map = {
+            401: ClientAuthenticationError,
+            404: ResourceNotFoundError,
+            409: ResourceExistsError,
+            304: ResourceNotModifiedError,
+        }
+    
+    def _build_sync_pipeline_client(self, auth_policy):
+        sync_core_policies = {
+            "UserAgentPolicy": UserAgentPolicy,
+            "HeadersPolicy": HeadersPolicy,
+            "ProxyPolicy": ProxyPolicy,
+            "NetworkTraceLoggingPolicy": NetworkTraceLoggingPolicy,
+            "HttpLoggingPolicy": HttpLoggingPolicy,
+            "RetryPolicy": RetryPolicy, 
+            "CustomHookPolicy": CustomHookPolicy,
+            "RedirectPolicy": RedirectPolicy,
+            "ContentDecodePolicy": ContentDecodePolicy,
+            "DistributedTracingPolicy": DistributedTracingPolicy,
+            "RequestIdPolicy": RequestIdPolicy,
+            "SensitiveHeaderCleanupPolicy": SensitiveHeaderCleanupPolicy,
+        } # last 4 are not included in default generated policies
+        default_policies = [
+            UserAgentPolicy,
+            HeadersPolicy,
+            ProxyPolicy,
+            NetworkTraceLoggingPolicy,
+            HttpLoggingPolicy,
+            RetryPolicy, 
+            CustomHookPolicy,
+            RedirectPolicy
+        ]
+
+        if self.args.policies is None:
+            # if None, only auth policy is passed in
+            sync_pipeline = Pipeline(
+                transport=self.sync_transport(),
+                policies=[auth_policy]
+            )
+        elif self.args.policies == "all":
+            # if all, autorest default policies + auth policy
+            sync_policies = [policy(sdk_moniker=self.sdk_moniker) for policy in default_policies]
+            sync_policies.append(auth_policy)
+            sync_pipeline = Pipeline(
+                transport=self.sync_transport(),
+                policies=sync_policies
+            )
+        else:
+            # if custom list of policies, pass in custom list + auth policy
+            try:
+                custom_policies = ast.literal_eval(self.args.policies)
+            except ValueError as exc:
+                raise ValueError(f"""Invalid policies list: {self.args.policies}\n."""
+                                 f"""Must be string list of string policies from the following list:\n"""
+                                 f"""{list(sync_core_policies.keys())}""") from exc
+            try:
+                sync_policies = []
+                for policy in custom_policies:
+                    # if ContentDecodePolicy passed in, check for kwargs for that policy
+                    if policy == "ContentDecodePolicy" and self.args.content_decode_policy_kwargs:
```

Comment:
What special kwargs does this policy have that makes this needed? Are we able to use a sensible default for now?

Reply:
Though I'm wondering if this could actually be made universal with something like:
```python
if hasattr(self.args, f"{policy.__name__}Kwargs":
    policy_kwargs = ...
```
Though I'm not entirely sure if argparse allows arbitrary arguments like this... and it would be annoying to have to actually define each of these...

Reply:
hmm, it wasn't working with binary data perf tests earlier for some reason but it's working now. I don't think any extra kwargs need to be passed in here anymore so will remove.